compaction/compaction.cc

```diff
@@ -1815,6 +1815,13 @@ static future<compaction_result> scrub_sstables_validate_mode(sstables::compacti
     };
 }
 
+future<compaction_result> scrub_sstables_validate_mode(sstables::compaction_descriptor descriptor, compaction_data& cdata, table_state& table_s, compaction_progress_monitor& progress_monitor = default_compaction_progress_monitor()) {
+    progress_monitor.set_generator(std::make_unique<compaction_read_monitor_generator>(table_s, use_backlog_tracker::no));
+    auto res = co_await scrub_sstables_validate_mode(descriptor, cdata, table_s, *progress_monitor._generator);
```

Comment:
auto d = defer([&] { progress_monitor.reset_generator() })  before running scrub_sstables_validate_mode() so we reset generator on exception.

Reply:
what about the success path?
Should we cancel the deferred action so the progress monitor will remain as 100%?
Or maybe it doesn't matter?